rero_ils/modules/api.py

```diff
@@ -565,9 +580,41 @@ def _index_action(self, payload):
             '_source': body
         }
         action.update(arguments)
-
+        self.after_index_record(record=record)
         return action
 
+    def _delete_action(self, payload):
+        """Bulk delete action.
+
+        :param payload: Decoded message body.
+        :returns: Dictionary defining an Elasticsearch bulk 'delete' action.
+        """
+        kwargs = {}
+        record = None
+        index, doc_type = payload.get('index'), payload.get('doc_type')
+        if not (index and doc_type):
+            record = self.record_cls.get_record(
+                payload['id'], with_deleted=True)
+            index, doc_type = self.record_to_index(record)
+            kwargs['_version'] = record.revision_id
+            kwargs['_version_type'] = self._version_type
+        else:
+            # Allow version to be sent in the payload (but only use if we
+            # haven't loaded the record.
+            if 'version' in payload:
+                kwargs['_version'] = payload['version']
+                kwargs['_version_type'] = self._version_type
+        index, doc_type = self._prepare_index(index, doc_type)
+        if record:
+            self.after_bulk_delete_record(record=record)
```

Comment:
I do not understand. The idea of bulk action is to do several actions at the same time. Here it seems that we delete only on record.

Reply:
This is the `delete action` for one record in an bulk delete. See https://invenio-indexer.readthedocs.io/en/latest/_modules/invenio_indexer/api.html